youtube_dl/extractor/sbs.py

```diff
@@ -46,9 +79,197 @@ class SBSIE(InfoExtractor):
     }, {
         'url': 'https://www.sbs.com.au/ondemand/watch/1698704451971',
         'only_matching': True,
+    }, {
+        'url': 'https://www.sbs.com.au/ondemand/movie/coherence/1469404227931',
+        'only_matching': True,
+    }, {
+        'note': 'Live stream',
+        'url': 'https://www.sbs.com.au/ondemand/video/1726824003663/sbs-24x7-live-stream-nsw',
+        'only_matching': True,
+    }, {
+        'url': 'https://www.sbs.com.au/ondemand/news-series/dateline/dateline-2022/dateline-s2022-ep26/2072245827515',
+        'only_matching': True,
+    }, {
+        'url': 'https://www.sbs.com.au/ondemand/tv-series/the-handmaids-tale/season-5/the-handmaids-tale-s5-ep1/2065631811776',
+        'only_matching': True,
     }]
 
+    def __handle_request_webpage_error(self, err, video_id=None, errnote=None, fatal=True):
+        if errnote is False:
+            return False
+        if errnote is None:
+            errnote = 'Unable to download webpage'
+
+        errmsg = '%s: %s' % (errnote, error_to_compat_str(err))
+        if fatal:
+            raise ExtractorError(errmsg, sys.exc_info()[2], cause=err, video_id=video_id)
+        else:
+            self._downloader.report_warning(errmsg)
+            return False
+
+    def _download_webpage_handle(self, url, video_id, *args, **kwargs):
+        # note, errnote, fatal, encoding, data, headers, query, expected_status
+        # specialised to detect geo-block
+
+        errnote = args[2] if len(args) > 2 else kwargs.get('errnote')
+        fatal = args[3] if len(args) > 3 else kwargs.get('fatal')
+        exp = args[7] if len(args) > 7 else kwargs.get('expected_status')
+
+        # add 403 to expected codes for interception
+        exp = variadic(exp or [], allowed_types=(compat_str, ))
+        if 403 not in exp and '403' not in exp:
+            exp = list(exp)
+            exp.append(403)
+        else:
+            exp = None
+
+        if exp:
+            if len(args) > 7:
+                args = list(args)
+                args[7] = exp
+            else:
+                kwargs['expected_status'] = exp
+                kwargs = compat_kwargs(kwargs)
+
```

Comment:
Address https://github.com/ytdl-org/youtube-dl/pull/31880#discussion_r1153294487 (not required for yt-dlp or future yt-dl)
```suggestion

        # for now, ensure that headers passed is not None
        if len(args) > 5:
            if args[5] is None:
                # replace with empty dict
                args = list(args)
                args[5] = {}
        else:
            if kwargs.get('headers', False) is None:
                # enforce callee default
                del kwargs['headers']

```